### DISCUSSION TITLE
Is it possible to run GUI apps using this WSL distro? If yes, then how?

### DISCUSSION THREAD

**Discussion by ezxpro** [sileshn/EndeavourOSWSL2#8] [Q&A]
I'm trying to do just this but I'm having an issue. I _think_ I'll have a problem trying to run xorg apps because there's no driver for the Microsoft vGPU in the arch repos, as far as I know.

But that might be just the tip of the iceberg tbh...

What do you think?

**Comment by sileshn** [comment]
By GUI apps, you mean a single app or the full desktop environment? A gui app should work ootb. You can see a screenshot of gedit running from EndeavourOS below. A full desktop experience is also possible, but involves a lot more steps.

<img width="960" alt="Screenshot 2024-01-12 151708" src="https://github.com/sileshn/EndeavourOSWSL2/assets/977360/6c5452b7-eb96-4e31-9b4c-999f26c702ce">

**Reply by ezxpro** [reply]
Hi, thanks for the answer.

I want to run a single application, the evolution mail client. However, when I try to do it, I get an error that's apparently related to the vGPU:

```
ezxpro@PORTAL ~ % evolution
MESA: error: ZINK: vkCreateInstance failed (VK_ERROR_INCOMPATIBLE_DRIVER)
libEGL warning: egl: failed to create dri2 screen
libEGL warning: MESA-LOADER: failed to open vgem: /usr/lib/dri/vgem_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/dri, suffix _dri)

libEGL warning: MESA-LOADER: failed to open vgem: /usr/lib/dri/vgem_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/dri, suffix _dri)

libEGL warning: failed to open /dev/dri/card0: Permission denied

[1]    253 segmentation fault (core dumped)  evolution
139 ezxpro@PORTAL ~ %                                                                     :(
```

**Reply by sileshn** [reply]
Strange, because it works for me. 

<img width="960" alt="Screenshot 2024-01-13 073823" src="https://github.com/sileshn/EndeavourOSWSL2/assets/977360/1266268b-beec-4b78-9ace-75d5869f6fb5">

Which GPU are you using? You may need to install more packages based on your GPU.

**Comment by sileshn** [comment]
Try installing mesa, dbus, mesa-vdpau, libva-mesa-driver libva-utils. Shutdown and restart distro
